### PR TITLE
Add ucontext for linux-musl.

### PR DESCRIPTION
**Pull request by fiveop**
It just copy pasted from the linux-other version.

### CODE REVIEW

**Comment by fiveop** [comment]
Given that musl certainly exports these methods

    $ git grep getcontext # in a clone of musl's master branch
    include/ucontext.h:int  getcontext(struct __ucontext *);

and `ucontext.h` is imported on linux indiscriminately (see lib-test\build.rs), why does the build fail because of undefined references to this (and similar) functions ( https://travis-ci.org/rust-lang/libc/jobs/121508994 )?

**Comment by alexcrichton** [comment]
I believe that while declared in MUSL the functions aren't actually defined.

**Comment by fiveop** [comment]
You are right: http://wiki.musl-libc.org/wiki/Open_Issues#ucontext.h . But It does provide the struct, so I just removed the functions declarations.

File: src/unix/notbsd/linux/musl/b64/x86_64.rs [hunk 1]
```diff
@@ -1,3 +1,18 @@
 pub const SYS_gettid: ::c_long = 186;
 
 pub const SYS_perf_event_open: ::c_long = 298;
+
+s! {
```
Comment (alexcrichton) [inline src/unix/notbsd/linux/musl/b64/x86_64.rs]: Currently structs come before constants, but otherwise this looks good to me!

**Comment by fiveop** [comment]
done

**Pull request merged**